telemetry-summarize/action.yml

```diff
@@ -1,110 +1,108 @@
-              repo: context.repo.repo,
-              run_id: context.runId
-            });
-
-            // We know what the run ID is, but we don't know which specific job we're being run from.
-            // https://github.com/orgs/community/discussions/8945
-            var this_job = job_info.data.jobs.find((job) => {
-              return job.runner_name === process.env.RUNNER_NAME && job.run_attempt === runAttempt;
-            });
-
-            return this_job;
-          };
-
-          var this_job = undefined;
-          var retryCount = 0;
-          maxRetries = 5;
-          initialDelay = 2000;
-          while (this_job === undefined) {
-            this_job = await get_info({github, context, process});
-            if (this_job === undefined && retryCount < maxRetries) {
-              retryCount++;
-              const delay = initialDelay * Math.pow(2, retryCount - 1);
-            } else {
-              break;
-            }
-          }
-
-          if (this_job === undefined){
-            throw "Telemetry values were not available. Please see debug logs for more info."
-          }
-          core.setOutput("job_info_json", this_job);
-
+        ca_cert: "${{inputs.ca_cert}}"
+        client_cert: "${{inputs.client_cert}}"
+        client_key: "${{inputs.client_key}}"
+    - uses: ./shared-actions/telemetry-set-otel-service-name
     - shell: bash
       run: |
-        echo ${{toJSON(steps.get-job-info.outputs.job_info_json)}} > job_info.json
-
         # This traceparent is the one that comes from outside; from any external job above this
         # Our job span should be a child of that trace. It is valid for this to be empty.
         export TRACEPARENT="${{inputs.traceparent}}"
 
-        export job_traceparent="$(rapids-get-telemetry-traceparent $job_name)"
-        # TODO: Pass through attrs and friends. Maybe this is easier to do by just
-        # setting the env var, though.
-        job_name="$(cat job_info.json | jq '.name')"
+        # NOTE: OTEL_SERVICE_NAME is set at this point. It either comes in as an env var set outside
+        #    of this action, or it is set in the `telemetry-set-otel-service-name` action above this.
+
+        export job_traceparent="$(rapids-get-telemetry-traceparent "${OTEL_SERVICE_NAME}")"
+
+        # Nest the steps under the job we just made a span for. This sets the parent span ID for us.
+        export TRACEPARENT=${job_traceparent}
+
+        # The reporting of the completion time is earlier than the actual last step's completion.
+        # We compensate for that by picking up the last known completion time of any step.
+        last_timestamp=$(date +'%s')
+
+        # Base64 encoding is to avoid issues with spaces/newlines/whatever funny business
+        for row in $(cat job_info.json | jq -r '.steps[] | @base64'); do
+            name="$(echo ${row} | base64 --decode | jq -r ".name")"
+            conclusion="$(echo ${row} | base64 --decode | jq -r ".conclusion")"
```

Comment:
Just wanted to point out that `jq` has a `@base64d` function as well: https://jqlang.github.io/jq/manual/#format-strings-and-escaping

![image](https://github.com/user-attachments/assets/851cab2d-2939-43b8-bcd5-3aff1b656832)

Makes no differences to me whether you change these lines, just wanted to mention it.